NeuroID/Utilities/NIDNetworkServiceImpl.swift

```diff
@@ -10,23 +10,34 @@ import Alamofire
 
 class NIDNetworkServiceImpl: NIDNetworkServiceProtocol {
 
-    func retryableRequest(url: URL, neuroHTTPRequest: NeuroHTTPRequest, headers: HTTPHeaders, retryCount: Int, completion: @escaping (AFDataResponse<Data>) -> Void) {
+    private var afCustomSession: Alamofire.Session
+    private let configuration = URLSessionConfiguration.af.default
+    
+    init() {
+    
+        // Initialize the session
+        self.afCustomSession = Alamofire.Session(configuration: configuration)
+    }
+
+    func retryableRequest(url: URL, neuroHTTPRequest: NeuroHTTPRequest, headers: HTTPHeaders, retryCount: Int = 0, completion: @escaping (AFDataResponse<Data>) -> Void) {
+        let maxRetryCount = 3
```

Comment:
This is just moving the max retry count down a layer into the network class rather than sending it in when the function is called?

Reply:
correct